src/engine/sparqlExpressions/SparqlExpression.h

```diff
@@ -89,7 +89,7 @@ class SparqlExpression {
 
   // Check if an expression is distinct (only applies to aggregates)
   virtual bool isDistinct() const {
-    AD_CONTRACT_CHECK(this->isAggregate());
+    AD_CONTRACT_CHECK(isAggregate());
```

Comment:
The lack of coverage of one of the branches in the `CONTRACT_CHECK` tells us something interesting here:
This function should probably always throw.
1. If this is not an aggregate then this is rather pointless.
2. For an aggregate, it indicates that the function was not overridden properly by the AGGREGATE (as it might be distinct).

So maybe something like
(without any checks)
`AD_THROW("isDistinct() maybe only called for aggregate expressions. If this is an aggregate expression, then the implementation of isDistinct() is missing for this expression. Please report this to the developers)`.

You can check this with `AD_EXPECT_THROW_WITH_MESSAGE`.